src/coreclr/src/jit/lsrabuild.cpp

```diff
@@ -2317,6 +2359,62 @@ void LinearScan::buildIntervals()
                 pos->setRegOptional(true);
             }
         }
+        // Adjust heuristics for writeThru intervals.
+        if (compiler->compHndBBtabCount > 0)
+        {
+            VarSetOps::Iter iter(compiler, exceptVars);
+            unsigned        varIndex = 0;
+            while (iter.NextElem(&varIndex))
+            {
+                unsigned   varNum   = compiler->lvaTrackedToVarNum[varIndex];
+                LclVarDsc* varDsc   = compiler->lvaTable + varNum;
+                Interval*  interval = getIntervalForLocalVar(varIndex);
+                assert(interval->isWriteThru);
+                BasicBlock::weight_t weight = varDsc->lvRefCntWtd();
+
+                // We'd like to only allocate registers for EH vars that have enough uses
+                // to compensate for the additional registers being live (and for the possibility
+                // that we may have to insert an additional copy).
+                // However, we don't currently have that information available. Instead, we'll
+                // aggressively assume that these vars are defined once, at their first RefPosition.
+                //
+                RefPosition* firstRefPosition = interval->firstRefPosition;
+
+                // Incoming reg args are given an initial weight of 2 * BB_UNITY_WEIGHT
+                // (see lvaComputeRefCounts(); this may be reviewed/changed in future).
+                //
+                BasicBlock::weight_t initialWeight = (firstRefPosition->refType == RefTypeParamDef)
+                                                         ? (2 * BB_UNITY_WEIGHT)
+                                                         : blockInfo[firstRefPosition->bbNum].weight;
+                weight -= initialWeight;
+
+                // If the remaining weight is less than the initial weight, we'd like to allocate it only
+                // opportunistically, but we don't currently have a mechanism to do so.
+                // For now, we'll just avoid using callee-save registers if the weight is too low.
+                if (interval->preferCalleeSave)
+                {
+                    // The benefit of a callee-save register isn't as high as it would be for a normal arg.
+                    // We'll have at least the cost of saving & restoring the callee-save register,
+                    // so we won't break even until we have at least 4 * BB_UNITY_WEIGHT.
+                    // Given that we also don't have a good way to tell whether the variable is live
+                    // across a call in the non-EH code, we'll be extra conservative about this.
+                    // Note that for writeThru intervals we don't update the preferences to be only callee-save.
```

Comment:
I take it this is the matching bit of logic for line 1185?

Reply:
Yes, I'll add a pointer